content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -818,6 +818,74 @@ controller is tracking a Job using Pod finalizers by checking if the Job has the
 annotation `batch.kubernetes.io/job-tracking`. You should **not** manually add
 or remove this annotation from Jobs.
 
+### Pod Communication Within a Job Using Pod Hostnames
+Some jobs may benefit from using pod hostnames for pod networking rather than pod IPs. 
+For example, IndexedJobs automatically set the pod hostname to be in the format of 
+`${jobName}-${completionIndex}`, which can be used to deterministically determine
+pod hostnames and enable pod networking *without* needing to create a client connection to
+the Kubernetes control plane to obtain pod hostnames/IPs via API requests. This can be useful
+for use cases where pod networking is required but we don't want to depend on a network 
+connection with the Kubernetes API server.
+
+
+To enable pod-to-pod communication using pod hostnames, you must do the following:
+1. Set up a [headless service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services)
+with a valid label selector for the pods created by your job. This will trigger `kube-dns` to cache the hostnames of the pods running your job. One easy way to do this is to use the `job-name: <your-job-name>` selector.
+2. Update the template spec in your job with the following: `subdomain: <headless-svc-name>`
+   where `<headless-svc-name>` must match the name of your headless service
+   exactly. 
+
+Example:
+
+```yaml
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: headless-svc
+spec:
+  ports:
+  - port: 80
+    name: web
+  clusterIP: None # clusterIP must be None to create a headless service
+  selector:
+    app: nginx # must match job template spec label
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: web
+spec:
+  completions: 3
+  parallelism: 3
+  completionMode: Indexed
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      subdomain: headless-svc # has to match headless service name
+      restartPolicy: Never
+      containers:
+      - name: nginx
+        image: registry.k8s.io/nginx-slim:0.8
+        ports:
+        - containerPort: 80
+          name: web
+```
+
+After applying the example above, other pods can reach eachother over
+the network using: `<pod-hostname>.<headless-service-name>`. 
```

Comment:
Important to link to https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/ somewhere to learn more about DNS resolution in Kubernetes.

In particular, it's important to note that this name wouldn't work in None or Default DNS policy https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy

Reply:
Also, I just learned that mini-kube doesn't include DNS by default. So definitely worth mentioning as a requirement.

https://minikube.sigs.k8s.io/docs/handbook/addons/ingress-dns/

Reply:
You are one step ahead of me!

Reply:
Ok I've added a note that the `<pod-hostname>.<headless-service-name>` name format wouldn't work with DNS Policy of `None` or `Default`, and included the link to learn more about pod DNS policies.